pkg/snapshotterutil/socisource.go

```diff
@@ -0,0 +1,110 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Taken from https://github.com/awslabs/soci-snapshotter/blob/237fc956b8366e49927c84fcfee9a2defbb8f53c/fs/source/source.go
+// to avoid taking dependency, as maintainers do not wish to upgrade to containerd v2 yet.
+
+package snapshotterutil
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/labels"
+	ctdsnapshotters "github.com/containerd/containerd/pkg/snapshotters"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const (
+	// TargetSizeLabel is a label which contains layer size.
+	TargetSizeLabel = "containerd.io/snapshot/remote/soci.size"
+
+	// targetImageLayersSizeLabel is a label which contains layer sizes contained in
+	// the target image.
+	targetImageLayersSizeLabel = "containerd.io/snapshot/remote/image.layers.size"
+
+	// TargetSociIndexDigestLabel is a label which contains the digest of the soci index.
+	TargetSociIndexDigestLabel = "containerd.io/snapshot/remote/soci.index.digest"
+)
+
+// AppendDefaultLabelsHandlerWrapper makes a handler which appends image's basic
```

Comment:
Nit: rename the comment to match `SociAppendDefaultLabelsHandlerWrapper`

Reply:
Thanks, fixed.